ibis/backends/polars/compiler.py

```diff
@@ -619,8 +619,10 @@ def distinct(op):
 def count_star(op):
     if (where := op.where) is not None:
         condition = translate(where)
-        return condition.filter(condition).count()
-    return pl.count()
+        # TODO: clean up the casts and use schema.apply_to in the backend's
+        # execute method
+        return condition.filter(condition).count().cast(pl.Int64)
```

Comment:
I'll have another PR for this, but it's going to disable support for some kinds of joins because polars seems to only return one of the key columns.

Reply:
This comment is now outdated, resolving.